src/plans/quasi_newton_plan.jl

```diff
@@ -632,22 +657,17 @@ Generate a cautious update for either a matrix based or a limited memory based u
 [`QuasiNewtonMatrixDirectionUpdate`](@ref)
 [`QuasiNewtonLimitedMemoryDirectionUpdate`](@ref)
 """
-mutable struct QuasiNewtonCautiousDirectionUpdate{U} <:
+mutable struct QuasiNewtonCautiousDirectionUpdate{U,Tθ} <:
                AbstractQuasiNewtonDirectionUpdate where {
-    U<:Union{QuasiNewtonMatrixDirectionUpdate,QuasiNewtonLimitedMemoryDirectionUpdate{T}}
-} where {T<:AbstractQuasiNewtonUpdateRule}
+    U<:Union{QuasiNewtonMatrixDirectionUpdate,QuasiNewtonLimitedMemoryDirectionUpdate}
+}
     update::U
-    θ::Function
+    θ::Tθ
```

Comment:
Thanks for spotting, this is definetly a code line from before I was aware how all struct parameters work :)

Reply:
(I miss comments that are basically resolved by default but mentioned to provide a short comment)

Reply:
Non-concrete field types is something I sometimes look for and try to avoid or remove if possible :)

Reply:
Yes very good; especially in older code of mine – or students of mine – where I (nor them) was not yet aware of that.